### PR TITLE
Feature/biometrics

### PR DESCRIPTION
**Pull request by TheBlueHawk**
This PR add biometric authentication to confirm the change of health status on supported phones.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/BiometricUtils.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                return false;
+            case BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE:
+                Log.e("MY_APP_TAG", "Biometric features are currently unavailable.");
+                return false;
+            case BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED:
+                Log.e("MY_APP_TAG", "The user hasn't associated " +
+                        "any biometric credentials with their account.");
+                return false;
+        }
+        return false;
```
[the remark this answers is not in the document]
Comment (TheBlueHawk) [reply app/src/main/java/ch/epfl/sdp/BiometricUtils.java:45]: Structure simplified with fever returns

File: app/src/main/java/ch/epfl/sdp/BiometricUtils.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        switch (biometricManager.canAuthenticate()) {
+            case BiometricManager.BIOMETRIC_SUCCESS:
+                Log.d("MY_APP_TAG", "App can authenticate using biometrics.");
+                return true;
+            case BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE:
+                Log.e("MY_APP_TAG", "No biometric features available on this device.");
+                return false;
+            case BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE:
+                Log.e("MY_APP_TAG", "Biometric features are currently unavailable.");
+                return false;
```
[the remark this answers is not in the document]
Comment (TheBlueHawk) [reply app/src/main/java/ch/epfl/sdp/BiometricUtils.java]: Structure simplified with fever returns

File: app/src/main/java/ch/epfl/sdp/BiometricUtils.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            case BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED:
+                Log.e("MY_APP_TAG", "The user hasn't associated " +
+                        "any biometric credentials with their account.");
+                return false;
```
[the remark this answers is not in the document]
Comment (TheBlueHawk) [reply app/src/main/java/ch/epfl/sdp/BiometricUtils.java]: Structure simplified with fever returns

File: app/src/main/java/ch/epfl/sdp/UserInfectionActivity.java [hunk 4]
```diff
@@ -29,31 +41,86 @@ protected void onCreate(Bundle savedInstanceState) {
...
+    public void onClick(View view) {
+        if (BiometricUtils.canAuthenticate(this)) {
+            biometricPrompt.authenticate(promptInfo);
+        } else {
+            executeHealthStatusChange();
+        }
+    }
 
+
+    private BiometricPrompt biometricPromptBuilder(Executor executor) {
```
[the remark this answers is not in the document]
Comment (TheBlueHawk) [reply app/src/main/java/ch/epfl/sdp/UserInfectionActivity.java]: Method size reduced by extracting private methods for each outcome of the callback

**Review by matteopariset — APPROVED** [review]
It would be very nice if GridFirestoreInteractor could implement FirestoreInteractor

**Review by AdrienRem — APPROVED** [review]
Ok even if test coverage is bad. It seems to be impossible to test that

**Pull request merged**